src/main/java/seedu/address/commons/util/RidePageGenerator.java

```diff
@@ -0,0 +1,63 @@
+package seedu.address.commons.util;
+
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.h2;
+import static j2html.TagCreator.img;
+import static j2html.TagCreator.p;
+
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+import j2html.tags.ContainerTag;
+import seedu.address.model.ride.Ride;
+
+/**
+ * Singleton Class for generating a Html page used to display ride details.
+ */
+public class RidePageGenerator extends HtmlGenerator<Ride> {
```

Comment:
This is great, makes it really easy to create ride pages haha, but would it make it hard to style the html pages?

Reply:
Yes it is difficult to style HTML pages still. I think HtmlGenerator can be used to hold common templates that can be reused, such as making horizontal tables.